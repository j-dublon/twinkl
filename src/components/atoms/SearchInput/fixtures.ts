import { SearchInputProps } from "@/types";
import { Mock } from "vitest";

export const mockOnChange: Mock = vi.fn();

export const mockProps: SearchInputProps = {
  value: "",
  onChange: mockOnChange,
};
