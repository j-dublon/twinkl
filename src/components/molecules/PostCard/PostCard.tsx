import { PostCardProps } from "@/types";
import { FC } from "react";

export const PostCard: FC<PostCardProps> = ({ post, removePost }) => (
  <div
    className="m-4 pb-1 focus:outline-none focus:ring-2 focus:ring-blue-500 border-b-1 border-b-zinc-300 flex flex-row justify-between items-center"
    tabIndex={0}
    aria-label="Post content"
  >
    <div>
      <h1>{post.title}</h1>
      <p>{post.body}</p>
    </div>
    <button
      className="text-xl text-red-600 hover:text-red-900 ml-16"
      onClick={() => removePost(post.id)}
      aria-label={`Remove ${post.title} post`}
    >
      X
    </button>
  </div>
);
