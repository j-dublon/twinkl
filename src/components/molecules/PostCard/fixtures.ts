import { Post } from "@/types";

export const mockPost: Post = {
  id: 1,
  title: "A great post",
  body: "Some information about this post",
  userId: 1,
};
