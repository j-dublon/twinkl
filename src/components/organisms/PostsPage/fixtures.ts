import { Mock } from "vitest";

export const mockDeletePost: Mock = vi.fn();
