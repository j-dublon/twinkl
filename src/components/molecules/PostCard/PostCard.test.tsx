import { render } from "@testing-library/react";
import { PostCard } from "./PostCard";
import { mockPost } from "./fixtures";

describe("Component: PostCard", () => {
  it("SHOULD match snapshot", () => {
    const component = render(<PostCard post={mockPost} />);
    expect(component).toMatchSnapshot();
  });

  it("SHOULD render text content as expected WHEN props are provided", () => {
    const { getByText } = render(<PostCard post={mockPost} />);
    expect(getByText(mockPost.title));
    expect(getByText(mockPost.body));
  });
});
