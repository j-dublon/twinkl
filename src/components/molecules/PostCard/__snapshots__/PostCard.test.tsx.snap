// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Component: PostCard > SHOULD match snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        aria-label="Post content"
        class="m-4 pb-1 focus:outline-none focus:ring-2 focus:ring-blue-500 border-b-1 border-b-zinc-300"
        tabindex="0"
      >
        <h1>
          A great post
        </h1>
        <p>
          Some information about this post
        </p>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      aria-label="Post content"
      class="m-4 pb-1 focus:outline-none focus:ring-2 focus:ring-blue-500 border-b-1 border-b-zinc-300"
      tabindex="0"
    >
      <h1>
        A great post
      </h1>
      <p>
        Some information about this post
      </p>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
